src/header/map.rs

```diff
@@ -3218,7 +3228,13 @@ fn usable_capacity(cap: usize) -> usize {
 
 #[inline]
 fn to_raw_capacity(n: usize) -> usize {
-    n + n / 3
+    match n.checked_add(n / 3) {
+        Some(n) => n,
+        None => panic!(
+            "requested capacity {} too large: overflow while converting to raw capacity",
+            n
+        ),
+    }
```

Comment:
TBH, this could just be:

```rust
n.checked_add(n / 3).expect("requested capacity must fit usize without overflow")
```

The only downside if that the requested capacity is not shown in the error, so maybe you'll want to hear what the maintainers think before applying any changes.

(the same applies to the above `match` that basically hand-rolls an `expect`.